src/MCPServer/lib/linkTaskManagerReplacementDicFromChoice.py

```diff
@@ -46,27 +46,33 @@ def __init__(self, jobChainLink, pk, unit):
         for i, dic in enumerate(dicts):
             self.choices.append((i, dic.description, dic.replacementdic))
 
-        # This is an attempt to find a choice in DashboardSettings for the
-        # chain link that the current MicroServiceChoiceReplacementDic is
-        # taking us. We're looking up the setting dict by the module name
-        # that the StandardTaskConfig wants to execute, e.g. upload-qubit_v0.0.
+        # There are MicroServiceChoiceReplacementDic links with no
+        # replacements (``self.choices`` has zero elements at this point). This
+        # is true for the following chain links:
         #
-        # Not implemented yet, but this could enable us to store more than a
-        # configuration set a specific chain link.
+        #   - ``Choose Config for Archivists Toolkit DIP Upload``
+        #   - ``Choose config for AtoM DIP upload``
+        #   - ``Choose Config for ArchivesSpace DIP Upload``
         #
-        # DashboardSettings does not belong to MCPServer. We currently have
-        # direct access to the database but that may not be always possible.
-        try:
-            mscl = MicroServiceChainLink.objects.get(id=jobChainLink.pk)
-            task_id = mscl.defaultnextchainlink.currenttask.tasktypepkreference
-            stc = StandardTaskConfig.objects.get(id=task_id)
-        except (MicroServiceChainLink.DoesNotExist, StandardTaskConfig.DoesNotExist, AttributeError):
-            pass
-        else:
-            args = DashboardSetting.objects.get_dict(stc.execute)
-            if args:
-                args = {'%{}%'.format(key): value for key, value in args.items()}
-                self.choices.append((len(self.choices), stc.execute, str(args)))
+        # The only purpose of these links is to  load settings from the
+        # Dashboard configuration store (``DashboardSetting``), e.g.
+        # connection details or credentials that are needed to perform the
+        # upload of the DIP to the remote system.
+        #
+        # Once the settings are loaded, we proceed with the next chain link
+        # automatically instead of prompting the user with a single choice
+        # which was considered inconvenient and confusing. In the future, it
+        # should be possible to prompt the user only if we want to have the
+        # user decide between multiple configurations, e.g. more than one
+        # AtoM instance is available and the user wants to decide which one is
+        # going to be used.
+        rdict = self._get_dashboard_setting_choice()
+        if rdict and not self.choices:
+            LOGGER.debug('Found Dashboard settings for this task, proceed.')
+            self.update_passvar_replacement_dict(rdict)
+            self.jobChainLink.linkProcessingComplete(
+                0, passVar=self.jobChainLink.passVar)
+            return
```

Comment:
This is the new magic. Basically, if we know there's just a single choice we just ask the engine to go with it instead of prompting the user - that's why we're calling `linkProcessingComplete` and return immediately.

Reply:
So, the dashboard look up can't return multiple choices at the moment, it's just a dict with a single choice in every case, right?

Reply:
Exactly, just a single dict that becomes part of `passVar` with all the values that the next script needs to upload the DIP to the access system, e.g. `%host%`, `%user%`, etc...